src/Microsoft.ML.StandardLearners/Standard/Online/AveragedPerceptron.cs

```diff
@@ -60,8 +60,13 @@ public sealed class Options : AveragedLinearOptions
             /// <summary>
             /// A custom <a href="tmpurl_loss">loss</a>.
             /// </summary>
-            [Argument(ArgumentType.Multiple, HelpText = "Loss Function", ShortName = "loss", SortOrder = 50)]
-            public ISupportClassificationLossFactory LossFunction = new HingeLoss.Options();
+            [Argument(ArgumentType.Multiple, Name = "LossFunction", HelpText = "Loss Function", ShortName = "loss", SortOrder = 50)]
+            internal ISupportClassificationLossFactory LossFunctionFactory1 = new HingeLoss.Options();
```

Comment:
>LossFunctionFactory1 [](start = 55, length = 20)

Was the `1` here intentional?

Reply:
I guess it was... would it be possible to give it a more clear name? As far as I can tell the choice of `1` was just an arbitrary value so that it didn't conflict with the existing `LossFunctionFactory` property, but ...

---
In reply to: [263109093](https://github.com/dotnet/machinelearning/pull/2856#discussion_r263109093) [](ancestors = 263109093)